profiler/include/profiler/profile_contraction_impl.hpp

```diff
@@ -137,20 +138,41 @@ int profile_contraction_impl(int do_verification,
                                                                       DataType,
                                                                       DataType,
                                                                       AElementOp,
-                                                                      BElementOp,
-                                                                      CDElementOp,
-                                                                      DataType>;
+                                                                      BElementOp>;
 
         auto ref_op      = ReferenceGemmInstance{};
         auto ref_invoker = ref_op.MakeInvoker();
 
-        if constexpr(std::is_same<CDElementOp, Scale>::value)
-            d_m_n = Tensor<DataType>(std::vector<ck::index_t>{}, std::vector<ck::index_t>{});
+        Tensor<DataType> c_m_n_host_result(f_host_tensor_descriptor(M, N, StridesC));
 
-        auto ref_argument = ref_op.MakeArgument(
-            a_m_k, b_k_n, d_m_n, c_m_n_host_result, a_element_op, b_element_op, cd_element_op);
+        auto ref_argument =
+            ref_op.MakeArgument(a_m_k, b_k_n, c_m_n_host_result, a_element_op, b_element_op);
 
         ref_invoker.Run(ref_argument);
+
+        for(size_t m0 = 0; m0 < e_m_n_host_result.mDesc.GetLengths()[0]; ++m0)
+        {
+            for(size_t m1 = 0; m1 < e_m_n_host_result.mDesc.GetLengths()[1]; ++m1)
+            {
+                for(size_t n0 = 0; n0 < e_m_n_host_result.mDesc.GetLengths()[2]; ++n0)
+                {
+                    for(size_t n1 = 0; n1 < e_m_n_host_result.mDesc.GetLengths()[3]; ++n1)
+                    {
+                        if constexpr(is_same<CDElementOp, Bilinear>::value)
+                        {
+                            cde_element_op(e_m_n_host_result(m0, m1, n0, n1),
+                                           c_m_n_host_result(m0, m1, n0, n1),
+                                           d_m_n(m0, m1, n0, n1));
+                        }
+                        else if constexpr(is_same<CDElementOp, Scale>::value)
+                        {
+                            cde_element_op(e_m_n_host_result(m0, m1, n0, n1),
+                                           c_m_n_host_result(m0, m1, n0, n1));
+                        }
```

Comment:
You haven't guarded here the last `else` condition when we do not support provided `CDElementOp`.

Reply:
Added.